sap-file-server-md/azuredeploy.json

```diff
@@ -0,0 +1,493 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "resourcePrefix": {
+      "type": "string",
+      "maxLength": 6,
+      "minLength": 3,
+      "metadata": {
+        "description": "The prefix that should be used to create the resource names."
+      }
+    },
+    "osType": {
+      "type": "string",
+      "allowedValues": [
+        "Windows Server 2016 Datacenter",
+        "SLES 12",
+        "SLES 12 BYOS"
+      ],
+      "defaultValue": "Windows Server 2016 Datacenter",
+      "metadata": {
+        "description": "The type of the operating system you want to deploy."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "minLength": 12,
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "newOrExistingSubnet": {
+      "type": "string",
+      "defaultValue": "new",
+      "allowedValues": [
+        "new",
+        "existing"
+      ],
+      "metadata": {
+        "description": "Determines whether a new virtual network and subnet should be created or an existing subnet should be used."
+      }
+    },
+    "subnetId": {

```

Comment:
use the condition property to determine when to deploy the resource - if this is empty, deploy the network - and the you can update the description for instruction... e.g. enter it subnetId to use, if blank a new virtualNetwork will be provisioned.